src/kernels/jupyter/jupyterDetectionTelemetry.node.ts

```diff
@@ -0,0 +1,112 @@
+        this.detectJupyter('lab', process.env).catch(noop);
+        const { env, shell } = await this.terminalHelper.getEnvironmentVariables(undefined);
+        if (!env) {
+            return;
+        }
+        const mergedVariables = { ...process.env };
+        Object.keys(env).forEach((key) => {
+            if (key in process.env) {
+                return;
+            }
+            if (key.toLowerCase() == 'path') {
+                // Append path from terminal.
+                let delimiter = path.delimiter;
+                const currentPath = (process.env['PATH'] || process.env['Path'] || '').trim();
+                const terminalPath = (env['PATH'] || env['Path'] || '').trim();
+                if (currentPath.endsWith(delimiter)) {
+                    delimiter = '';
+                }
+                if (terminalPath.startsWith(delimiter)) {
+                    delimiter = '';
+                }
+                mergedVariables[key] = `${currentPath}${delimiter}${terminalPath}`;
+            } else {
+                mergedVariables[key] = env[key];
+            }
+        });
+        this.detectJupyter('notebook', mergedVariables, shell).catch(noop);
+        this.detectJupyter('lab', mergedVariables, shell).catch(noop);
+    }
+    private async detectJupyter(
+        frontEnd: 'notebook' | 'lab',
+        env: NodeJS.ProcessEnv,
+        shell?: TerminalShellType
+    ): Promise<void> {
+        try {
+            const processService = await this.processFactory.create(undefined);
+            const output = await processService.exec('jupyter', ['notebook', '--version'], {
+                env: env || process.env,
+                throwOnStdErr: false,
+                mergeStdOutErr: true
+            });
+            const versionLines = output.stdout
+                .splitLines({ trim: true, removeEmptyEntries: true })
+                .filter((line) => !isNaN(parseInt(line.substring(0, 1), 10)));
+            const versionMatch = /^\s*(\d+)\.(\d+)\.(.+)\s*$/.exec(versionLines.length ? versionLines[0] : '');
+            if (versionMatch && versionMatch.length > 2) {
+                const major = parseInt(versionMatch[1], 10);
+                const minor = parseInt(versionMatch[2], 10);
+                const frontEndVersion = parseFloat(`${major}.${minor}`);
+                if (shell) {
+                    sendTelemetryEvent(Telemetry.JupyterInstalled, undefined, {
+                        frontEnd,
+                        frontEndVersion,
+                        detection: 'shell',
+                        shellType: shell
+                    });
+                } else {
+                    sendTelemetryEvent(Telemetry.JupyterInstalled, undefined, {
+                        frontEnd,
+                        frontEndVersion,
+                        detection: 'process'
+                    });
+                }
+            }
+            throw new Error('Not installed');
```

Comment:
doesn't look like this should be here, I don't see a return above this.

Reply:
However there is a catch which will handle this error.
Hence this exception, to go into the catch clause.

Reply:
yeah, I'm saying it shouldn't always go to that catch clause, there isn't another option right now.

Reply:
Not following you, but i've removed the error and copied & pasted the code to send the telemetry.